pkg/csi/service/common/util.go

```diff
@@ -371,3 +378,49 @@ func Contains(list []string, item string) bool {
 	}
 	return false
 }
+
+// GetClusterComputeResourceMoIds helps find ClusterComputeResourceMoIds from
+// AvailabilityZone CRs on the supervisor cluster
+func GetClusterComputeResourceMoIds(ctx context.Context) ([]string, error) {
+	// Get a config to talk to the apiserver.
+	clusterComputeResourceMoIds := make([]string, 0)
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
+	}
+
+	// Create a new AvailabilityZone client.
+	azClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AvailabilityZone client using config. Err: %+v", err)
+	}
+	azResource := schema.GroupVersionResource{
+		Group: "topology.tanzu.vmware.com", Version: "v1alpha1", Resource: "availabilityzones"}
+	// Get AvailabilityZone list
+	azList, err := azClient.Resource(azResource).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		// If the AvailabilityZone CR is not registered in the
+		// supervisor cluster, we receive NoKindMatchError. In such cases
+		// return nil array for ClusterComputeResourceMoIds with no error
+		// AvailabilityZone CR is not registered when supervisor cluster was installed prior to vSphere 8.0
+		// Upgrading vSphere to 8.0 does not create zone CRs on existing supervisor clusters
+		_, ok := err.(*apiMeta.NoKindMatchError)
+		if ok {
+			log.Infof("AvailabilityZone CR is not registered on the cluster")
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to get AvailabilityZone lists. err: %+v", err)
+	}
+	if len(azList.Items) == 0 {
```

Comment:
If the AZ CR is registered in the environment, is it necessary that a CR instance should be available?

Reply:
yes, it should be present. The older supervisor will not have AZ cr registered.
only new supervisor clusters will have AZ CR registered and at least one CR instance is present.